src/openrct2/peep/GuestPathfinding.cpp

```diff
@@ -2073,11 +2073,11 @@ int32_t guest_path_finding(Guest* peep)
 
         int16_t stationX = (int16_t)(entranceLocation.x * 32);
         int16_t stationY = (int16_t)(entranceLocation.y * 32);
-        uint16_t dist = abs(stationX - peep->next_x) + abs(stationY - peep->next_y);
-
-        if (dist < closestDist)
+        uint16_t score = abs(stationX - peep->next_x) + abs(stationY - peep->next_y);
+        score += abs(entranceLocation.z - peep->next_z) * 2;
```

Comment:
Any reason only z is multiple of two? If I recall correctly calculating the distance between 2 points in 3d space goes something like
```
sqrt(pow((x - x2), 2) + pow((y - y2), 2) + pow((z - z2), 2));
```
Given we want integers we can just skip sqrt

Reply:
Its to match the other calculation of the score.

Reply:
Perhaps because peeps walk slower on sloped paths? A comment to explain the logic behind the equations would be helpful.

Reply:
Instead of score/distance I would also consider this a "cost" if the effort is included in the mix.

Reply:
Its to bring it more inline with this calc.
```
        // Calculate the heuristic score of this map element.
        uint16_t x_delta = abs(gPeepPathFindGoalPosition.x - loc.x) * 32;
        uint16_t y_delta = abs(gPeepPathFindGoalPosition.y - loc.y) * 32;
        if (x_delta < y_delta)
            x_delta >>= 4;
        else
            y_delta >>= 4;
        uint16_t new_score = x_delta + y_delta;
        uint16_t z_delta = abs(gPeepPathFindGoalPosition.z - loc.z);
        z_delta <<= 1;
        new_score += z_delta;
```

Reply:
So we have essentially duplicate code for calculating a cost between two points in 3d space where Z has a bigger impact on the cost.

Reply:
I always assumed it did Manhattan distance for performance reasons.

Reply:
As for path finding what the actual metric is doesn't matter, its just a cost/weight value. Question was more about the multiple of two, if the reason is for making it consistent with the other code then we should deduplicate that logic and perhaps mention that slopes are more expensive cost wise in the function (assuming that is the reason behind it)

Reply:
Okay. Its now been changed into just the one score generator for the two functions.